**Importation des bibliothèques**

In [ ]:
import tensorflow
from tensorflow.keras import backend as k
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout,Conv1D, MaxPooling1D,Flatten,BatchNormalization
from sklearn.model_selection import train_test_split
import numpy
import pandas as pd
import matplotlib.pyplot as plt

from tensorflow.keras.callbacks import ModelCheckpoint, LearningRateScheduler


import tensorflow as tf
print(tf.__version__)

import numpy as np
print(np.__version__)

numpy.random.seed(2)
import pandas as pd

2.15.0
1.26.4


**Configurer les GPUs**

In [ ]:
config = tensorflow.compat.v1.ConfigProto()
sess = tensorflow.compat.v1.Session(config=config)


**Importation de la base**

In [ ]:
from google.colab import files

uploaded = files.upload()

for fn in uploaded.keys():
  print('User uploaded file "{name}" with length {length} bytes'.format(
      name=fn, length=len(uploaded[fn])))

Saving FS1.txt to FS1.txt
Saving profile.txt to profile.txt
Saving PS2.txt to PS2.txt
User uploaded file "FS1.txt" with length 7581926 bytes
User uploaded file "profile.txt" with length 31196 bytes
User uploaded file "PS2.txt" with length 82373611 bytes


**Préparation de la base FS1**

In [ ]:
FE_1= pd.read_csv("FS1.txt",delimiter="	",header=None)
FE_1=numpy.array(FE_1,float)

X_e1 = FE_1[:,:]
#m=np.max(X_e1)
print(X_e1.shape)

(2205, 600)


**Préparation de la base PS2**

In [ ]:
FE_2= pd.read_csv("PS2.txt",delimiter="	",header=None)
FE_2=numpy.array(FE_2,float)

X_e2 = FE_2[:,:]
print(X_e2.shape)

(2205, 6000)


**Préparation de la base Profile**

In [ ]:
dataset_output= pd.read_csv("profile.txt",delimiter="	",header=None)
dataset_output=numpy.array(dataset_output,float)
dataset_output = dataset_output[:,1]
dataset_output_new= dataset_output.reshape(dataset_output.shape[0], )
print(dataset_output_new.shape)

(2205,)


**Visualisation de la base FS1**

In [ ]:
df1 = pd.DataFrame(X_e1)
df1


0      1      2      3      4      5      6      7      8      9    \
0     8.990  0.770  0.641  0.006  0.000  0.000  0.001  0.003  0.001  0.001   
1     8.919  0.815  0.709  0.009  0.004  0.000  0.001  0.000  0.000  0.001   
2     9.179  0.683  0.528  0.008  0.003  0.001  0.003  0.003  0.004  0.006   
3     9.034  0.728  0.595  0.009  0.001  0.004  0.003  0.003  0.003  0.001   
4     8.729  0.705  0.446  0.014  0.007  0.003  0.001  0.003  0.001  0.000   
...     ...    ...    ...    ...    ...    ...    ...    ...    ...    ...   
2200  8.173  1.024  0.834  0.009  0.000  0.001  0.000  0.003  0.001  0.000   
2201  8.317  0.883  0.799  0.008  0.005  0.000  0.000  0.000  0.000  0.001   
2202  8.170  0.686  0.583  0.011  0.000  0.003  0.004  0.000  0.000  0.000   
2203  8.496  0.543  0.366  0.009  0.005  0.000  0.000  0.000  0.001  0.003   
2204  8.373  0.776  0.667  0.006  0.000  0.003  0.000  0.000  0.000  0.000   

      ...    590    591    592    593    594    595    596    597    598  \
0     ...  7.743  7.992  7.919  7.773  7.955  7.823  7.963  7.876  7.738   
1     ...  7.831  8.003  7.923  7.745  7.867  7.747  7.969  7.969  7.963   
2     ...  7.862  7.815  7.894  7.743  7.936  7.770  7.982  7.873  7.898   
3     ...  7.631  7.949  7.773  8.054  7.827  8.011  7.919  7.938  7.877   
4     ...  7.771  7.936  7.919  7.946  7.804  7.983  7.838  7.882  7.894   
...   ...    ...    ...    ...    ...    ...    ...    ...    ...    ...   
2200  ...  7.904  7.777  7.938  7.841  7.741  8.004  7.869  7.719  8.003   
2201  ...  7.766  7.957  7.817  7.739  8.036  7.871  7.647  7.993  7.898   
2202  ...  7.898  7.648  7.953  7.871  7.859  7.921  7.770  7.982  7.859   
2203  ...  7.834  7.737  8.065  7.893  7.653  7.971  7.898  7.758  7.892   
2204  ...  7.866  7.716  8.068  7.861  7.707  8.029  7.814  7.783  7.943   

        599  
0     8.036  
1     7.890  
2     7.952  
3     7.773  
4     7.825  
...     ...  
2200  7.874  
2201  7.787  
2202  7.685  
2203  7.793  
2204  7.774  

[2205 rows x 600 columns]

**Visualisation de la base PS2**

In [ ]:
df2 = pd.DataFrame(X_e2)
df2

0       1       2       3       4       5       6       7       8     \
0     125.50  125.39  125.40  125.03  124.05  123.18  104.01  56.500  23.992   
1     125.06  125.08  125.09  124.69  123.84  123.14  103.63  63.687  28.359   
2     125.13  125.27  125.23  124.74  123.94  123.23  106.35  60.516  26.258   
3     124.93  124.96  124.92  124.41  123.60  122.88  103.99  58.859  27.781   
4     124.72  124.74  124.66  124.31  123.57  122.74  105.94  62.648  30.875   
...      ...     ...     ...     ...     ...     ...     ...     ...     ...   
2200  125.47  125.63  125.70  125.34  124.94  124.88  113.56  71.586  34.516   
2201  125.83  125.70  125.59  125.43  125.18  124.98  114.30  72.531  35.234   
2202  125.41  125.49  125.63  125.38  124.96  124.77  113.69  70.773  32.531   
2203  125.91  125.73  125.59  125.38  125.10  124.96  114.19  74.039  38.695   
2204  125.88  125.86  125.84  125.48  125.03  124.87  113.13  72.687  38.437   

        9     ...    5990    5991    5992    5993    5994    5995    5996  \
0     18.406  ...  125.02  125.00  125.10  125.09  124.98  124.91  124.98   
1     21.711  ...  124.80  124.88  125.13  125.22  125.09  124.98  125.06   
2     19.258  ...  124.61  124.69  124.74  124.71  124.59  124.64  124.74   
3     21.469  ...  124.82  124.79  124.69  124.69  124.77  124.83  124.69   
4     23.883  ...  124.80  124.67  124.49  124.56  124.69  124.62  124.45   
...      ...  ...     ...     ...     ...     ...     ...     ...     ...   
2200  25.531  ...  125.57  125.51  125.38  125.43  125.59  125.61  125.53   
2201  26.820  ...  125.86  125.77  125.74  125.81  125.77  125.66  125.58   
2202  23.773  ...  125.66  125.73  125.77  125.81  125.94  126.04  125.98   
2203  30.125  ...  125.55  125.56  125.58  125.74  125.84  125.73  125.66   
2204  30.586  ...  125.48  125.60  125.65  125.60  125.63  125.73  125.81   

        5997    5998    5999  
0     125.11  125.14  125.10  
1     125.13  125.09  125.04  
2     124.73  124.77  124.88  
3     124.53  124.51  124.59  
4     124.41  124.47  124.51  
...      ...     ...     ...  
2200  125.50  125.60  125.73  
2201  125.58  125.55  125.44  
2202  125.84  125.84  125.94  
2203  125.73  125.86  125.91  
2204  125.77  125.66  125.77  

[2205 rows x 6000 columns]

**Visualisation de la base Profile**

In [ ]:
label = pd.DataFrame(dataset_output)
label

0
0     100.0
1     100.0
2     100.0
3     100.0
4     100.0
...     ...
2200  100.0
2201  100.0
2202  100.0
2203  100.0
2204  100.0

[2205 rows x 1 columns]

**Affecter "1" pour le cycle à 100 sinon "0"**

In [ ]:
Y=label[0]
for i in range (len(Y)):
  if (Y[i]==100):
    Y[i]=1
  else:
    Y[i]=0

**Combiner les entrées, résultat une base de 6600 features et 2205 exemples**

In [ ]:
nsamples, nx = df1.shape
nsamples, ny = df2.shape

dataset=np.zeros((nsamples,nx+ny))
print(dataset.shape)

dataset[:,0:nx]=df1
dataset[:,nx:]=df2
print(dataset.shape)
print(dataset[2204])

(2205, 6600)
(2205, 6600)
[  8.373   0.776   0.667 ... 125.77  125.66  125.77 ]


**Répartition entre l'apprentissage et test**

In [ ]:
x_train=dataset[:2000]

In [ ]:
x_test=dataset[2000:]

In [ ]:
y_train=Y[:2000]
y_test=Y[2000:]

In [ ]:
print(x_train.shape)
print(x_test.shape)
print(y_train.shape)
print(y_test.shape)

(2000, 6600)
(205, 6600)
(2000,)
(205,)


**Préparation des sorties du modèle**

**Préparation des entrées du modèle**

**Construction du modèle**

### **Tester le Random Forest Classifier pour la classifcation**

In [ ]:
from sklearn.ensemble import RandomForestClassifier


In [ ]:
rfc = RandomForestClassifier(n_estimators=150,criterion='gini',
                             random_state=42, min_samples_split = 4,
                             min_samples_leaf= 3, max_features = "log2",
                             oob_score = True, verbose = 9,
                             class_weight = "balanced",
                             min_weight_fraction_leaf = 0.00,
                             n_jobs = 10, warm_start = True,
                             )

rfc.fit(x_train,y_train)

/usr/local/lib/python3.10/dist-packages/sklearn/ensemble/_forest.py:861: UserWarning: class_weight presets "balanced" or "balanced_subsample" are not recommended for warm_start if the fitted data differs from the full dataset. In order to use "balanced" weights, use compute_class_weight ("balanced", classes, y). In place of y you can use a large enough sample of the full training set target to properly estimate the class frequency distributions. Pass the resulting weights as the class_weight parameter.
  warn(
[Parallel(n_jobs=10)]: Using backend ThreadingBackend with 10 concurrent workers.
[Parallel(n_jobs=10)]: Done  12 tasks      | elapsed:    0.0s
[Parallel(n_jobs=10)]: Done  30 tasks      | elapsed:    0.1s
[Parallel(n_jobs=10)]: Done  52 tasks      | elapsed:    0.1s


building tree 1 of 150
building tree 2 of 150
building tree 3 of 150
building tree 4 of 150
building tree 5 of 150
building tree 6 of 150
building tree 7 of 150building tree 8 of 150
building tree 9 of 150
building tree 10 of 150

building tree 11 of 150
building tree 12 of 150
building tree 13 of 150
building tree 14 of 150
building tree 15 of 150
building tree 16 of 150
building tree 17 of 150
building tree 18 of 150
building tree 19 of 150building tree 20 of 150
building tree 21 of 150

building tree 22 of 150building tree 23 of 150
building tree 24 of 150

building tree 25 of 150
building tree 26 of 150
building tree 27 of 150
building tree 28 of 150
building tree 29 of 150
building tree 30 of 150
building tree 31 of 150
building tree 32 of 150
building tree 33 of 150building tree 34 of 150
building tree 35 of 150
building tree 36 of 150
building tree 37 of 150

building tree 38 of 150
building tree 39 of 150
building tree 40 of 150
building tree 41 of 150
building tree 42 of 150
b

[Parallel(n_jobs=10)]: Done  78 tasks      | elapsed:    0.2s
[Parallel(n_jobs=10)]: Done 108 tasks      | elapsed:    0.2s
[Parallel(n_jobs=10)]: Done 148 out of 150 | elapsed:    0.3s remaining:    0.0s
[Parallel(n_jobs=10)]: Done 150 out of 150 | elapsed:    0.3s finished


building tree 108 of 150
building tree 109 of 150building tree 110 of 150

building tree 111 of 150building tree 112 of 150

building tree 113 of 150
building tree 114 of 150
building tree 115 of 150building tree 116 of 150

building tree 117 of 150
building tree 118 of 150
building tree 119 of 150
building tree 120 of 150
building tree 121 of 150
building tree 122 of 150
building tree 123 of 150
building tree 124 of 150
building tree 125 of 150
building tree 126 of 150
building tree 127 of 150
building tree 128 of 150
building tree 129 of 150
building tree 130 of 150building tree 131 of 150

building tree 132 of 150
building tree 133 of 150
building tree 134 of 150
building tree 135 of 150
building tree 136 of 150
building tree 137 of 150
building tree 138 of 150
building tree 139 of 150
building tree 140 of 150building tree 141 of 150

building tree 142 of 150
building tree 143 of 150
building tree 144 of 150
building tree 145 of 150building tree 146 of 150

building tree 147 of 150


RandomForestClassifier(class_weight='balanced', max_features='log2',
                       min_samples_leaf=3, min_samples_split=4,
                       n_estimators=150, n_jobs=10, oob_score=True,
                       random_state=42, verbose=9, warm_start=True)

**Evaluer les perfromances de test**

In [ ]:
predicted_output = rfc.predict(x_test)
print(predicted_output)
print(y_test)

[0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.
 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 1. 0. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.
 1. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 0. 0. 0. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 0. 0. 0. 1. 1. 1. 1. 0. 1. 1. 0. 1. 0. 1. 1. 1. 1. 1. 1.
 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
2000    0.0
2001    0.0
2002    0.0
2003    0.0
2004    0.0
       ... 
2200    1.0
2201    1.0
2202    1.0
2203    1.0
2204    1.0
Name: 0, Length: 205, dtype: float64


[Parallel(n_jobs=10)]: Using backend ThreadingBackend with 10 concurrent workers.
[Parallel(n_jobs=10)]: Done  12 tasks      | elapsed:    0.0s
[Parallel(n_jobs=10)]: Done  30 tasks      | elapsed:    0.0s
[Parallel(n_jobs=10)]: Done  52 tasks      | elapsed:    0.0s
[Parallel(n_jobs=10)]: Done  78 tasks      | elapsed:    0.0s
[Parallel(n_jobs=10)]: Done 108 tasks      | elapsed:    0.0s
[Parallel(n_jobs=10)]: Done 148 out of 150 | elapsed:    0.0s remaining:    0.0s
[Parallel(n_jobs=10)]: Done 150 out of 150 | elapsed:    0.0s finished


**Construire la matrice de confusion et générer le rapport de classification**

In [ ]:
from sklearn.metrics import confusion_matrix,classification_report
print (confusion_matrix(y_test,predicted_output))
print(classification_report(y_test,predicted_output))

[[124   8]
 [  0  73]]
              precision    recall  f1-score   support

         0.0       1.00      0.94      0.97       132
         1.0       0.90      1.00      0.95        73

    accuracy                           0.96       205
   macro avg       0.95      0.97      0.96       205
weighted avg       0.96      0.96      0.96       205

